### PR TITLE
Add useGeolocationEnabled hook and use public query

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Extracts the logic to show the geolocation input on checkout to the hook `useGeolocationEnabled`.

#### How should this be manually tested?

[Workspace](https://lucas--checkoutio.myvtex.com/cart/add?sku=289).

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/useGeolocationEnabled.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { useQuery } from 'react-apollo'
+
+import QueryGeolocationInstalled from './QueryGeolocationInstalled'
+
+export default function useGeolocationEnabled() {
```
Comment (wdsrocha) [inline react/useGeolocationEnabled.tsx:5]: Nit: how about using a simple export (without default)? My reasons are the same as the ones mentioned [here](https://basarat.gitbook.io/typescript/main-1/defaultisbad).

**Review by wdsrocha — APPROVED** [review]
nice

File: react/useGeolocationEnabled.tsx (hunk 1 above)
Comment (lucasecdb) [reply react/useGeolocationEnabled.tsx:5]: we can't use simple exports from an IO app entrypoint

Comment (wdsrocha) [reply react/useGeolocationEnabled.tsx:5]: just out of curiosity, this is a limitation with IO, right?

Comment (lucasecdb) [reply react/useGeolocationEnabled.tsx:5]: yes

**Pull request merged**